frontend/src/components/PipelineSpecTabContent.tsx

```diff
@@ -31,9 +31,14 @@ export function PipelineSpecTabContent(props: PipelineSpecTabContentProps) {
     <Editor
       value={
         isTemplateV2(props.templateString)
-          ? jsyaml.safeDump(jsyaml.safeLoad(props.templateString || ''))
-          : props.templateString || ''
-      } // Use safeLoad and then safeDump to make sure the v2 PipelineSpec is in Yaml Form.
+          ? props.templateString || ''
+          : jsyaml.safeDump(jsyaml.safeLoad(props.templateString || ''))
```

Comment:
How about just removing the V2 template check?

`value={jsyaml.safeDump(jsyaml.safeLoad(props.templateString || '')}`

We can keep the comments below to explain difference between V1 and V2.

Reply:
If we remove the v2 template check, we shouldn't say "Directly use"(?)